### ISSUE TITLE
Publish CFE precompiled to Kernel to cloud storage to unlock bootstrapping without prebuilt SDK

### ISSUE THREAD

**Issue by mraleph** [dart-lang/sdk#51788]
Currently we rely on prebuilt Dart SDK to bootstrap, however in reality we only need CFE part of it as long as CFE is compiled to Kernel binary compatible with the current VM version. 

During bootstrap we can build a `run-kernel` binary (essentially `dart` without `kernel-service`, `vm-service` or other core libraries snapshot included). `run-kernel` + `prebuilt-cfe.dill` will give us equivalent of `dart`, e.g. `dart input.dart` is done through `run-kernel prebuilt-cfe.dill input.dart /tmp/output.dill && run-kernel /tmp/output.dill`. Which will allow us to complete the bootstrap.

**Comment by sortie** [comment]
Is there anything we can help with here?

I could e.g. set up a trusted builder that makes these dill files and puts them in the cloud storage and cipd alongside the official releases? Then we can amend the checked in SDK mechanism with this alternative

**Comment by mraleph** [comment]
Yeah, I think we need some infrastructure e.g. decide where we put these and what kind of bots we need around this. Somebody on the VM team can do the GN / scripts wiring then.
